legacy/reqcounter/reqcounter.go

```diff
@@ -16,30 +16,73 @@ limitations under the License.
 
 package reqcounter
 
-import "sync"
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
 
 // RequestCounter records the number of HTTP requests to GCR.
 type RequestCounter struct {
-	mutex    sync.Mutex
-	requests uint64
+	mutex    sync.Mutex // Lock to prevent race-conditions with concurrent processes.
+	requests uint64     // Number of HTTP requests since recording started.
+	quota    uint64     // Maximum requests allowed before GCR throttling occurs.
+	since    time.Time  // When the current counter began recording requests.
+}
+
+// log records the number of HTTP requests found and resets the counter.
+func (rc *RequestCounter) log() {
+	// Grab the lock to prevent race-conditions.
+	rc.mutex.Lock()
+	defer rc.mutex.Unlock()
```

Comment:
I created `log` and `reset` as methods for RequestCounter because they are unique to the counter itself. Therefore, it has built in functions to modify itself. Having these also simplifies the `requestLogger` function.

Reply:
Ohhhh! I see your point, because I was still reading `counter.requests`. Fixing!